src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -90,6 +91,36 @@ public class SqlQueryEngine : IQueryEngine
             }
         }
 
+        /// <summary>
+        public async Task<Tuple<JsonDocument?, IMetadata?>> ExecuteAsync(IMiddlewareContext context, List<IDictionary<string, object?>> parameters, string dataSourceName)
+        {
+
+            string entityName = GraphQLUtils.GetEntityNameFromContext(context);
+
+            SqlQueryStructure structure = new(
+                context,
+                parameters,
+                _sqlMetadataProviderFactory.GetMetadataProvider(dataSourceName),
+                _authorizationResolver,
+                _runtimeConfigProvider,
+                _gQLFilterParser,
+                new IncrementingInteger(),
+                entityName);
+
+            if (structure.PaginationMetadata.IsPaginated)
+            {
+                return new Tuple<JsonDocument?, IMetadata?>(
+                    SqlPaginationUtil.CreatePaginationConnectionFromJsonDocument(await ExecuteAsync(structure, dataSourceName, isMultipleCreateOperation: true), structure.PaginationMetadata),
+                    structure.PaginationMetadata);
+            }
+            else
+            {
+                return new Tuple<JsonDocument?, IMetadata?>(
+                    await ExecuteAsync(structure, dataSourceName, isMultipleCreateOperation: true),
```

Comment:
Do you mean `isCreateMultipleOperation` rather than `isMultipleCreateOperation`? (since the latter can represent both point and multiple insertions.)

Reply:
Although this is invoked only for many type multiple create operation, I've named the parameter to be `isMultipleCreateOperation` to indicate whether this is a follow-up query executed as part of a multiple create operation or a query operation.